bitcoin/src/lib.rs

```diff
@@ -880,6 +893,49 @@ impl BitcoinCoreApi for BitcoinCore {
         Ok(())
     }
 
+    async fn rescan_electrs_for_addresses<A: PartialAddress + Send + Sync + 'static>(
+        &self,
+        addresses: Vec<A>,
+    ) -> Result<(), Error> {
+        for address in addresses.into_iter() {
+            let address = address.encode_str(self.network)?;
+            let all_transactions =
+                esplora_btc_api::apis::address_api::get_address_tx_history(&self.electrs_config, &address).await?;
+            // filter to only import a) payments in the blockchain (the API returns mempool transactions
+            // too) and b) payments TO the address (the API also returns any txes that have that
+            // address in the vin, which we will already have since we will have made the payment)
+            let confirmed_payments_to = all_transactions.into_iter().filter(|tx| {
+                if let Some(status) = &tx.status {
+                    if !status.confirmed {
+                        return false;
+                    }
+                };
+                if let Some(vout) = &tx.vout {
+                    for output in vout {
+                        if let Some(addr) = &output.scriptpubkey_address {
+                            if addr == &address {
+                                return true;
+                            }
+                        }
+                    }
+                };
+                false
```

Comment:
```suggestion
        // check that the address is used as _output_, since presumably bitcoin core already knows about transactions only _spend from_ the address 
        tx.vout
            .unwrap_or_default()
            .iter()
            .any(|output| matches!(output.scriptpubkey_address, Some(addr) if addr == address));
```